mbed_app.json

```diff
@@ -1,24 +1,27 @@
 {
     "macros": [
-        "MBED_TRACE_MAX_LEVEL=TRACE_LEVEL_INFO",
-        "MBED_CLIENT_USER_CONFIG_FILE=\"mbed_cloud_client_user_config.h\"",
-        "MBED_CLOUD_CLIENT_USER_CONFIG_FILE=\"mbed_cloud_client_user_config.h\"",
-        "ARM_UC_USE_PAL_BLOCKDEVICE=1",
         "PAL_PLATFORM_DEFINED_CONFIGURATION=\"mbedOS_SST.h\"",
         "PAL_USER_DEFINED_CONFIGURATION=\"mbedOS_SST.h\""
     ],
     "target_overrides": {
         "*": {
-            "target.features_add"                       : ["BOOTLOADER", "STORAGE"],
             "target.c_lib"                              : "std",
             "platform.stdio-baud-rate"                  : 115200,
             "platform.stdio-convert-newlines"           : true,
             "platform.stdio-buffered-serial"            : true,
             "platform.stdio-flush-at-exit"              : true,
             "rtos.main-thread-stack-size"               : 5120,
-            "update-client.storage-locations"           : 1,
-            "mbed-trace.enable"                         : null,
             "events.shared-stacksize"                   : 2048,
+            "mbed-trace.enable"                         : null,
+            "storage.storage_type"                      : null,
+            "storage_filesystem.filesystem"             : null,
+            "storage_filesystem.blockdevice"            : null,
+            "storage_tdb_internal.internal_base_address": null,
+            "storage_tdb_internal.internal_size"        : null,
+            "storage_filesystem.internal_base_address"  : null,
+            "storage_filesystem.rbp_internal_size"      : null,
+            "storage_filesystem.external_base_address"  : null,
+            "storage_filesystem.external_size"          : null,
             "nsapi.default-wifi-security"               : "WPA_WPA2",
             "nsapi.default-wifi-ssid"                   : "\"SSID\"",
             "nsapi.default-wifi-password"               : "\"PASSWORD\"",
```

Comment:
Changing those lines will change values also for the legacy targets, which you don't want.
Two solutions:
1. Config each target with the relevant params.
2. Move the legacy targets to a new file (`mbed_app_legacy.json`).

Reply:
A quick query here:

We observed that the configuration values of storage_tdb_* parameters are not getting picked from **mbed_lib.json** as defined in **prebuilt-bl** folder. Is this expected in case of FOTA configuration or are we missing something that needs to be added to support this.

Reply:
It should be taken from the mbed_lib.json of the prebuilt-bl folder
Are you seeing it for all devices?
I've went through the configuration here and can't find something obvious

Reply:
I've tried to compile the branch for K64F
I've changed the bootloader file name in the mbed_lib.json of the prebuilt-bl folder to a different name and had a compilation error. Seems to me that it takes the mbed_lib.json as expected

Reply:
For storage_tdb_* parameters we were observing that the values from the Mbed_lib.json were not getting picked. We verified that by checking the generated Mbed_config.h file.

Repo -> git@github.com:ARMmbed/mbed-os-example-pelion.git
Branch -> remotes/origin/fota_update

If you are available some time tomorrow we can have a short discussion at a suitable time and we can show you what we are observing.
Please let me know what time will be suitable for a short TEAMS meeting.

Reply:
@jenia81 

Thanks Jenia, we understand and we will experiment keeping what you said in focus.
We will update you about the same.

Reply:
@jenia81 

File .profile-c has all the MACROS as defined in mbed_lib.json.
There is an additional observation details of which are captured in the Excel sheet attached here.

[JSON_issue.xlsx](https://github.com/ARMmbed/mbed-os-example-pelion/files/7282707/JSON_issue.xlsx)

It seems that in case storage_tdb_* params are not defined in mbed_app.json there seems to be a issue in booting the board.

Reply:
@arpitarm Yes, it is true. The board won't boot if you will not define storage configuration parameters in mbed_app.json.
 `mbed-os` build system has this hierarchy order of config parameters: 
1. If application mbed_app.json has no config defines, the build system uses the default values of storage library that defined by the mbed_lib.json files. In case of storage it uses  [mbed_lib.json](https://github.com/ARMmbed/mbed-os/blob/master/storage/kvstore/kv_config/tdb_internal/mbed_lib.json) as default configs for internal storage. The board can't boot with these configs.
2. If application mbed_app.json uses `target_overrides` , the build system ignores the default target configs and uses application overrides.
So in case of storage configuration we disable default configurations by setting nulls in application mbed_app.json, and define storage macros by adding `target.macros_add` in prebuild mbed_lib.json.
You can see all macros in .profile-c file.

Please let us know if you have additional issues/questions.